go/interpreter/lfvm/memory.go

```diff
@@ -79,27 +83,24 @@ func (m *Memory) EnsureCapacity(offset, size uint64, c *context) error {
 		return errGasUintOverflow
 	}
 	if m.length() < needed {
-		needed = toValidMemorySize(needed)
 		fee := m.getExpansionCosts(needed)
 		if !c.useGas(fee) {
 			c.status = statusOutOfGas
 			return errOutOfGas
 		}
-		m.total_memory_cost += fee
-		m.store = append(m.store, make([]byte, needed-m.length())...)
+		m.expandMemoryWithoutCharging(needed)
 	}
+
 	return nil
 }
 
-func (m *Memory) EnsureCapacityWithoutGas(size uint64) {
-	if size <= 0 {
-		return
-	}
-	if m.length() < size {
-		size = toValidMemorySize(size)
-		fee := m.getExpansionCosts(size)
-		m.total_memory_cost += fee
-		m.store = append(m.store, make([]byte, size-m.length())...)
+// expandMemoryWithoutCharging expands the memory to the given size without charging gas.
+func (m *Memory) expandMemoryWithoutCharging(needed uint64) {
```

Comment:
nit: name feels odd. 

`expandMemoryWithoutChargingGas`
or 
`expandMemoryWithoutCharge`
..
`expandMemoryWithoutCost`
`expandMemoryWithoutGasCost`

Reply:
this is described in the documentation, so I will leave the name as is.